post: a Well-Known change password (#34)
cloudx-labs/posts: posts/navarroaxel/02.well-known-change-password.md

---
title: Build a discoverable change password form
published: true
description: How to create a discoverable change-password form in a website.
tags: 'webdev, security, ux, webperf'
id: 487021
---

We can use a well-known URL in our websites to make our _change password_ forms discoverable by tools, like 1Password, LastPass, Bitwarden, etc.

For `https://example.com`, this magic URL would be `https://example.com/.well-known/change-password`.

> Let's check how this works. 💪

Reviewer: ```suggestion
Let's see how this works. 💪
```

## A well-known URL

What is a [well-known URI](https://ldapwiki.com/wiki/Well-Known%20URIs)? It's basically a group of predefined "links" inside a `/.well-known/` directory of a website.

These links can be either a HTTP redirect, like the change-password URI, or a document with a specific format, like the "well known" OpenID Connect configuration in JSON format located at `/.well-known/openid-configuration`. We can find an example here <https://login.microsoftonline.com/common/v2.0/.well-known/openid-configuration>.

## Just a redirect

> The [specification](https://w3c.github.io/webappsec-change-password-url/) for Well-Known URLs for Change passwords are still in draft, but we can found examples in popular websites like [GitHub](https://github.com/.well-known/change-password), [Slack](https://slack.com/.well-known/change-password), or [Wikipedia](https://wikipedia.org/.well-known/change-password).

Author: :thinking: but `Change password` is the feature name. Perhaps this better:
```suggestion
The [specification](https://w3c.github.io/webappsec-change-password-url/) for Well-Known Change Password URL is still in draft, but we can found examples in popular websites like [GitHub](https://github.com/.well-known/change-password), [Slack](https://slack.com/.well-known/change-password), or [Wikipedia](https://wikipedia.org/.well-known/change-password).
```

> The [specification](https://w3c.github.io/webappsec-change-password-url/) for Well-Known URLs for Change passwords are still in draft, but we can found examples in popular websites like [GitHub](https://github.com/.well-known/change-password), [Slack](https://slack.com/.well-known/change-password), or [Wikipedia](https://wikipedia.org/.well-known/change-password).

Reviewer: Ah, then yes, that's better.

💡 If you can't find the reset password link in a website, you can use this Well-Known URL.

## Conclusion

> We can programmatically advertise where a user can change its password, improving the **security** and **usability** of a website that requires authentication.

Reviewer: ```suggestion
We can programmatically advertise where a user can change their password, improving the **security** and **usability** of a website that requires authentication.
```

> Now you can create a semantic HTML form for change passwords and the password manager tool will make the rest.

Reviewer: ```suggestion
Now you can create a semantic HTML form for changing passwords and the password manager tool will do the rest.
```
